aws/data_source_aws_launch_template.go

```diff
@@ -336,19 +336,35 @@ func dataSourceAwsLaunchTemplate() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
-			"tags": tagsSchemaComputed(),
+			"tags":   tagsSchemaComputed(),
+			"filter": dataSourceFiltersSchema(),
 		},
 	}
 }
 
 func dataSourceAwsLaunchTemplateRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	log.Printf("[DEBUG] Reading launch template %s", d.Get("name"))
+	filters, filtersOk := d.GetOk("filter")
+	name, nameOk := d.GetOk("name")
+	tags, tagsOk := d.GetOk("tags")
 
-	dlt, err := conn.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
-		LaunchTemplateNames: []*string{aws.String(d.Get("name").(string))},
-	})
+	if !filtersOk && !nameOk && !tagsOk {
```

Comment:
Can this now be done at `terraform plan` time with [`Schema.AtLeastOneOf`](https://github.com/hashicorp/terraform-plugin-sdk/blob/ac2f8d01136a1a0198e27d49ea0ea6ed986ea4ee/helper/schema/schema.go#L215-L228)?

Reply:
See [this comment](https://github.com/terraform-providers/terraform-provider-aws/pull/12404#discussion_r393003198).

Reply:
removed check entirely